### PR TITLE
fix: didn't show error comes from Rust

### PR DESCRIPTION
**Pull request by PavelLaptev**
fix of this issue https://github.com/gitbutlerapp/gitbutler/issues/3591

### CODE REVIEW

File: app/src/lib/notifications/toasts.ts [hunk 1]
```diff
@@ -22,7 +22,8 @@ export function showToast(toast: Toast) {
 }
 
 export function showError(title: string, err: any) {
```
Comment (Byron) [inline app/src/lib/notifications/toasts.ts:24]: Would it be possible to strongly type this `err`? Or is that prevented by `err` somehow being `string | Error`?

From all I can tell, the backend [should always provide a map with `code` and `message`](https://github.com/gitbutlerapp/gitbutler/blob/5a9962bc5a602ab932a36445715ccae03a2016c3/crates/gitbutler-tauri/src/error.rs#L60-L68), but apparently at least in this case, that doesn't happen.

Comment (Byron) [reply app/src/lib/notifications/toasts.ts:24]: Screenshot 2024-04-24 at 17 04 40

```
2024-04-24T15:04:26.171529Z ERROR add_project: crates/gitbutler-tauri/src/projects.rs:27: error=Error(Bare repositories are unsupported

Caused by:
    bare repositories are not supported) path="/Users/byron/dev-gb/bare"
```

This looks like the error type is not a string even when it arrives in the frontend.

Do errors come in from other places than [the IPC code here](https://github.com/gitbutlerapp/gitbutler/blob/dac89e41a980d3e5acfef50f624d8539d5f0079a/app/src/lib/backend/ipc.ts#L26-L31)?

**Review by Byron — COMMENTED** [review]
Thanks! It definitely works again.

Screenshot 2024-04-24 at 17 04 32

Please see my comments as well, I think there might be a more general issue with error handling and I'd love to fix the source of it. Ideally, the frontend can then refer to errors with a type as well *(which might be out of scope for this PR, of course)*.

**Comment by PavelLaptev** [comment]
Got it! Thanks for checking this @Byron. I'll go ahead and merge the PR to fix the bug asap. 
@mtsgrd, tagging you here in case you want to review @Byron's feedback as well.

**Pull request merged**